### PR TITLE
e2e: separate client certificates for BR

### PR DESCRIPTION
**Pull request by weekface**
### What problem does this PR solve? 
fixes: https://github.com/pingcap/tidb-operator/issues/2405
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -1345,6 +1184,15 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
...
 			framework.ExpectNoError(err)
 		})
+
+		ginkgo.It("Backup and Restore with TLS enabled", func() {
+			provider := framework.TestContext.Provider
+			if provider != "aws" && provider != "kind" {
+				framework.Skipf("provider is not aws or kind, skipping")
+			}
+
+			testBR("kind", ns, fw, c, genericCli, oa, cli, true)
```
Comment (cofyc) [inline tests/e2e/tidbcluster/tidbcluster.go:1194]:
```suggestion
			testBR(provider, ns, fw, c, genericCli, oa, cli, true)
```

if `kind` is only supported, you can skip if the provider is not `kind`

**Comment by shuijing198799** [comment]
no need test without tls ?

**Comment by weekface** [comment]
> no need test without tls ?

Here it is https://github.com/pingcap/tidb-operator/pull/2462/files#diff-92579a43c91e473b9716eeeee2068004R294

**Comment by shuijing198799** [comment]
> > no need test without tls ?
> 
> Here it is https://github.com/pingcap/tidb-operator/pull/2462/files#diff-92579a43c91e473b9716eeeee2068004R294

OK LGTM

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 2]
```diff
@@ -1487,3 +1329,202 @@ func newTidbClusterConfig(cfg *tests.Config, ns, clusterName, password, tidbVers
...
+	tcTo.Spec.TiKV.Replicas = 1
+	tcTo.Spec.TiDB.Replicas = 1
+	if tlsEnabled {
+		tcTo.Spec.TiDB.TLSClient = &v1alpha1.TiDBTLSClient{Enabled: true}
+	}
+	err = genericCli.Create(context.TODO(), tcTo)
+	framework.ExpectNoError(err)
+	err = oa.WaitForTidbClusterReady(tcTo, 30*time.Minute, 15*time.Second)
+	framework.ExpectNoError(err)
+	clusterTo := newTidbClusterConfig(e2econfig.TestConfig, ns, tcNameTo, "", utilimage.TiDBV3Version)
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:1419]: It's v4 in L1408, does it need to be v4 here?

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 3]
```diff
@@ -1487,3 +1329,202 @@ func newTidbClusterConfig(cfg *tests.Config, ns, clusterName, password, tidbVers
...
+	// check backup is successed
+	err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+		tmpBackup, err := cli.PingcapV1alpha1().Backups(ns).Get(backup.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		// Check the status in conditions one by one,
+		// if the status other than complete or failed is running
+		for _, condition := range tmpBackup.Status.Conditions {
+			if condition.Type == v1alpha1.BackupComplete {
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:1468]: Suggest checking both type and status here.

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 4]
```diff
@@ -1487,3 +1329,202 @@ func newTidbClusterConfig(cfg *tests.Config, ns, clusterName, password, tidbVers
...
+				return true, nil
+			} else if condition.Type == v1alpha1.BackupFailed {
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:1470]: Suggest checking both type and status here.

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 5]
```diff
@@ -1487,3 +1329,202 @@ func newTidbClusterConfig(cfg *tests.Config, ns, clusterName, password, tidbVers
...
+	// check restore is successed
+	err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+		tmpRestore, err := cli.PingcapV1alpha1().Restores(ns).Get(restore.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		// Check the status in conditions one by one,
+		// if the status other than complete or failed is running
+		for _, condition := range tmpRestore.Status.Conditions {
+			if condition.Type == v1alpha1.RestoreComplete {
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:1497]: Suggest checking both type and status here.

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 6]
```diff
@@ -1487,3 +1329,202 @@ func newTidbClusterConfig(cfg *tests.Config, ns, clusterName, password, tidbVers
...
+				return true, nil
+			} else if condition.Type == v1alpha1.RestoreFailed {
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:1499]: Suggest checking both type and status here.

**Pull request merged**